File: frameworks/motia/motia-py/playground/steps/dotenv_example/dotenv_check_step.py
"""Returns environment variables loaded from .env file."""

import os
from typing import Any

from motia import ApiRequest, ApiResponse, http, logger

config = {
    "name": "DotenvCheck",
    "description": "Returns environment variables loaded from .env file",
    "triggers": [
        http("GET", "/dotenv-check"),
    ],
    "enqueues": [],
    "flows": ["dotenv-example"],
}


def handler(request: ApiRequest[Any]) -> ApiResponse[dict[str, Any]]:
    """Return env vars loaded from .env."""
    greeting_prefix = os.environ.get("GREETING_PREFIX", "NOT_SET")
    app_name = os.environ.get("APP_NAME", "NOT_SET")
    has_secret_key = bool(os.environ.get("SECRET_KEY"))

    logger.info("Dotenv check", {"greeting_prefix": greeting_prefix, "app_name": app_name, "has_secret_key": has_secret_key})

    return ApiResponse(
        status=200,
        body={
            "greeting_prefix": greeting_prefix,
            "app_name": app_name,
            "has_secret_key": has_secret_key,
        },
    )

File: frameworks/motia/motia-js/playground/steps/dotenv-example/dotenv-check.step.ts
import { logger, type Handlers, type StepConfig } from 'motia'
import { z } from 'zod'

export const config = {
  name: 'DotenvCheck',
  description: 'Returns environment variables loaded from .env file',
  triggers: [
    {
      type: 'http',
      path: '/dotenv-check',
      method: 'GET',
      responseSchema: {
        200: z.object({
          greetingPrefix: z.string(),
          appName: z.string(),
          hasSecretKey: z.boolean(),
        }),
      },
    },
  ],
  enqueues: [],
  flows: ['dotenv-example'],
} as const satisfies StepConfig

export const handler: Handlers<typeof config> = async () => {
  const greetingPrefix = process.env.GREETING_PREFIX || 'NOT_SET'
  const appName = process.env.APP_NAME || 'NOT_SET'
  const hasSecretKey = !!process.env.SECRET_KEY

  logger.info('Dotenv check', { greetingPrefix, appName, hasSecretKey })

  return {
    status: 200,
    body: { greetingPrefix, appName, hasSecretKey },
  }
}
